### PR TITLE
[zkclient] Add user disable mechanism using unix socket.

### PR DESCRIPTION
**Pull request by marcopeereboom**

### CODE REVIEW

File: zkclient/zkclient.go [hunk 1]
```diff
@@ -1336,17 +1426,25 @@ func (z *ZKC) handleRPC() {
...
 				},
-				rpc.Empty{})
+				rpc.Acknowledge{})
 
 		case rpc.TaggedCmdAcknowledge:
+			var a rpc.Acknowledge
+			_, err = xdr.Unmarshal(br, &a)
+			if err != nil {
+				exitError = fmt.Errorf("unmarshal " +
+					"Acknowledge")
```
Comment (dajohi) [inline zkclient/zkclient.go]: : %v", err)

File: zkserver/rendezvous.go [hunk 2]
```diff
@@ -97,7 +97,7 @@ func (z *ZKS) handleRendezvousPull(writer chan *RPCWrapper,
 		goto bad
 	}
 	br = bytes.NewReader(rzXDR)
-	_, err = xdr.Unmarshal(br, &rzRecord)
+	_, err = z.unmarshal(br, &rzRecord)
 	if err != nil {
 		payload.Error = fmt.Sprintf("internal error unmarshal")
```
Comment (dajohi) [inline zkserver/rendezvous.go]: : %v", err)

File: zkserver/zkserver.go [hunk 3]
```diff
@@ -116,6 +127,36 @@ func (z *ZKS) writeMessage(kx *session.KX, msg *RPCWrapper) error {
...
+	payload := rpc.Unwelcome{
+		Version: rpc.ProtocolVersion,
+		Reason:  reason,
+	}
+
+	// encode command
+	var bb bytes.Buffer
+	_, err := xdr.Marshal(&bb, message)
+	if err != nil {
+		return fmt.Errorf("could not marshal Unwelcome message")
```
Comment (dajohi) [inline zkserver/zkserver.go]: : %v", err)

File: zkserver/zkserver.go [hunk 4]
```diff
@@ -116,6 +127,36 @@ func (z *ZKS) writeMessage(kx *session.KX, msg *RPCWrapper) error {
...
+	}
+	_, err = xdr.Marshal(&bb, payload)
+	if err != nil {
+		return fmt.Errorf("could not marshal Unwelcome payload")
```
Comment (dajohi) [inline zkserver/zkserver.go]: : %v", err)

Comment (marcopeereboom) [reply zkserver/zkserver.go]: XDR errors spit the input out. It is very loud and provides no utility to print it. For example, a chunk is 256KB and if we print it it'll be more than 1MB worth of text. No one wants to see that.

**Pull request merged**